diff --git a/src/Standard/Standard_ErrorHandler.hxx b/src/Standard/Standard_ErrorHandler.hxx
index 09bdc7bfbf..726951fb89 100644
--- a/src/Standard/Standard_ErrorHandler.hxx
+++ b/src/Standard/Standard_ErrorHandler.hxx
@@ -155,13 +155,13 @@ public:
     DEFINE_STANDARD_ALLOC
 
     //! Registers this callback object in the current error handler (if found).
-    void RegisterCallback();
+    Standard_EXPORT void RegisterCallback();
 
     //! Unregisters this callback object from the error handler.
-    void UnregisterCallback();
+    Standard_EXPORT void UnregisterCallback();
 
     //! Destructor
-    virtual ~Callback();
+    Standard_EXPORT virtual ~Callback();
 
     //! The callback function to perform necessary callback action.
     //! Called by the exception handler when it is being destroyed but
