diff --git a/cmake/FindXDR.cmake b/cmake/FindXDR.cmake
index e8e8625..a47508f 100644
--- a/cmake/FindXDR.cmake
+++ b/cmake/FindXDR.cmake
@@ -21,8 +21,16 @@ MESSAGE(STATUS "Check for XDR ...")
 
 FIND_PATH(XDR_INCLUDE_DIRS rpc/xdr.h)
 IF(XDR_INCLUDE_DIRS)
-   SET(XDR_DEFINITIONS "-DHAS_XDR")
-ENDIF()
+  SET(XDR_DEFINITIONS "-DHAS_XDR")
+ELSE(XDR_INCLUDE_DIRS)
+  IF(EXISTS "/usr/include/tirpc/rpc/xdr.h")
+    MESSAGE(STATUS "XDR was found in /usr/include/tirpc ...")
+    include_directories(/usr/include/tirpc)
+    SET(XDR_DEFINITIONS "-DHAS_XDR")
+    SET(XDR_INCLUDE_DIRS "/usr/include/tirpc")
+    FIND_LIBRARY(XDR_LIBRARIES tirpc)
+  ENDIF()
+ENDIF(XDR_INCLUDE_DIRS)
 
 IF(WIN32)
   FIND_LIBRARY(XDR_LIBRARIES xdr)                 # To get the .lib file from XDR
