diff --git a/src/GEOMImpl/GEOMImpl_ICurvesOperations.cxx b/src/GEOMImpl/GEOMImpl_ICurvesOperations.cxx
index 4727a5c5d..f9cff2743 100644
--- a/src/GEOMImpl/GEOMImpl_ICurvesOperations.cxx
+++ b/src/GEOMImpl/GEOMImpl_ICurvesOperations.cxx
@@ -21,14 +21,13 @@
 //
 
 #ifdef WIN32
-// E.A. : On windows with python 2.6, there is a conflict
-// E.A. : between pymath.h and Standard_math.h which define
-// E.A. : some same symbols : acosh, asinh, ...
-#include <Standard_math.hxx>
-#include <pymath.h>
+  #include <Standard_math.hxx>
+  #include <Python.h>
+  #include <pymath.h>
+#else
+  #include <Python.h>
 #endif
 
-#include <Python.h>
 #include <structmember.h>
 
 #ifdef HAVE_FINITE
