diff -Naur gmsh_ref/CMakeLists.txt gmsh_new/CMakeLists.txt
--- gmsh_ref/CMakeLists.txt	2022-05-26 10:51:23.000000000 +0200
+++ gmsh_new/CMakeLists.txt	2022-06-14 15:01:11.502544660 +0200
@@ -1340,9 +1340,6 @@
     set_config_option(HAVE_OCC "OpenCASCADE")
     list(APPEND EXTERNAL_LIBRARIES ${OCC_LIBS})
     list(APPEND EXTERNAL_INCLUDES ${OCC_INC})
-    if(WIN32 AND NOT MSVC)
-      add_definitions(-DOCC_CONVERT_SIGNALS)
-    endif()
   endif()
 endif()
 
