diff -Naur h5py/h5py/h5l.pyx h5py.dev/h5py/h5l.pyx
--- h5py/h5py/h5l.pyx	2021-05-09 13:55:36.000000000 +0200
+++ h5py.dev/h5py/h5l.pyx	2025-04-29 16:00:10.862670324 +0200
@@ -184,7 +184,7 @@
             if info.type == H5L_TYPE_SOFT:
                 py_retval = buf
             else:
-                H5Lunpack_elink_val(buf, buf_size, &wtf, &ext_file_name, &ext_obj_name)
+                H5Lunpack_elink_val(buf, buf_size, &wtf, <const char **>&ext_file_name, <const char **>&ext_obj_name)
                 py_retval = (bytes(ext_file_name), bytes(ext_obj_name))
         finally:
             efree(buf)
