diff -Naur libigl.ref/cmake/recipes/external/cgal.cmake libigl/cmake/recipes/external/cgal.cmake
--- libigl.ref/cmake/recipes/external/cgal.cmake	2024-12-17 15:55:31.000000000 +0100
+++ libigl/cmake/recipes/external/cgal.cmake	2026-04-24 09:36:23.737693400 +0200
@@ -1,3 +1,16 @@
+if (EXISTS $ENV{CGAL_DIR})
+    set(CGAL_DIR $ENV{CGAL_DIR})
+    list(APPEND CMAKE_PREFIX_PATH $ENV{CGAL_DIR})
+endif()
+
+find_package(CGAL QUIET COMPONENTS Core PATHS ${CGAL_DIR} NO_DEFAULT_PATH)
+if(CGAL_FOUND)
+    message(STATUS "Found CGAL: ${CGAL_DIR}")
+    return()
+else()
+    message(FATAL_ERROR "CGAL not found: ${CGAL_DIR}")
+endif()
+
 if(TARGET CGAL::CGAL)
     return()
 endif()
