diff -rupN matplotlib-1.4.2_old/setupext.py matplotlib-1.4.2_new/setupext.py
--- matplotlib-1.4.2_old/setupext.py	2014-10-23 03:54:01.000000000 +0200
+++ matplotlib-1.4.2_new/setupext.py	2014-12-12 14:43:21.000000000 +0100
@@ -448,6 +448,11 @@ class SetupPackage(object):
                     "Requires %s %s or later.  Found %s." %
                     (package, min_version, version))
 
+        if package=='freetype2':
+            include_dirs_hack = [os.path.join(os.environ['FREETYPE_ROOT_DIR'], 'include'),  os.path.join(os.environ['FREETYPE_ROOT_DIR'], 'include', 'freetype2') ]
+            check_include_file(include_dirs_hack, include_file, package)
+            return 'version %s' % version
+
         ext = self.get_extension()
         if ext is None:
             ext = make_extension('test', [])
@@ -984,18 +989,22 @@ class FreeType(SetupPackage):
                 return '.'.join([major, minor, patch])
 
     def add_flags(self, ext):
+        ## SALOME HACK
+        freetype_include_path = os.path.join(os.environ['FREETYPE_ROOT_DIR'], 'include')
+        freetype_include_freetype2_path = os.path.join(freetype_include_path, 'freetype2')
+        freetype_include_freetype2_freetype_path = os.path.join(freetype_include_path, 'freetype2', 'freetype')
+        freetype_lib_path = os.path.join(os.environ['FREETYPE_ROOT_DIR'], 'lib')
+
         pkg_config.setup_extension(
             ext, 'freetype2',
-            default_include_dirs=[
-                'include/freetype2', 'freetype2',
-                'lib/freetype2/include',
+            default_include_dirs=[freetype_include_path, freetype_include_freetype2_path, freetype_include_freetype2_freetype_path,
+                'freetype2', 'lib/freetype2/include',
                 'lib/freetype2/include/freetype2'],
-            default_library_dirs=[
+            default_library_dirs=[ freetype_lib_path,
                 'freetype2/lib'],
             default_libraries=['freetype', 'z'])
 
 
-
 class FT2Font(SetupPackage):
     name = 'ft2font'
 
