--- med-4.2.0_ref/src/hdfi/_MEDfileCreate.c	2025-05-15 17:20:19.000000000 +0200
+++ med-4.2.0_dev/src/hdfi/_MEDfileCreate.c	2025-07-16 17:16:42.420797585 +0200
@@ -79,7 +79,11 @@
     _h518medfileMMN[MED_FILE_H518_MINOR  ]= _i8_minor;
     _h518medfileMMN[MED_FILE_H518_RELEASE]= _i8_release; /* 0 */
 
-    _fp = fopen(filename, "w");
+#if defined(WIN32)
+    _fp = fopen(filename, "wb");
+#else
+	_fp = fopen(filename, "w");
+#endif
     if(!_fp) {
       MED_ERR_(_fid,MED_ERR_CREATE,MED_ERR_FILE,filename);
       ISCRUTE(major);ISCRUTE(minor);ISCRUTE(release);
