diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ccce7c..b032d22 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -240,15 +240,15 @@ endif(MEDCOUPLING_ENABLE_RENUMBER)
 
 if(MEDCOUPLING_ENABLE_SHAPERECOGN)
   if(UNIX)
-    set(CBLAS_ROOT_DIR
-        $ENV{CBLAS_ROOT_DIR}
-        CACHE PATH "Path to the LAPACK/CBLAS.")
-    if(CBLAS_ROOT_DIR)
-      list(APPEND CMAKE_PREFIX_PATH "${CBLAS_ROOT_DIR}")
-      find_package(CBLAS REQUIRED)
-    else(CBLAS_ROOT_DIR)
-      find_package(BLAS REQUIRED)
-    endif(CBLAS_ROOT_DIR)
+    set(OPENBLAS_ROOT_DIR
+        $ENV{OPENBLAS_ROOT_DIR}
+        CACHE PATH "Path to the OpenBLAS.")
+    if(OPENBLAS_ROOT_DIR)
+      list(APPEND CMAKE_PREFIX_PATH "${OPENBLAS_ROOT_DIR}")
+      find_package(OpenBLAS REQUIRED)
+    else(OPENBLAS_ROOT_DIR)
+      message(FATAL "Could not find OpenBLAS")
+    endif(OPENBLAS_ROOT_DIR)
     set(LAPACK_ROOT_DIR
         $ENV{LAPACK_ROOT_DIR}
         CACHE PATH "Path to the LAPACKE")
@@ -259,13 +259,13 @@ if(MEDCOUPLING_ENABLE_SHAPERECOGN)
       list(APPEND CMAKE_PREFIX_PATH "${LAPACK_ROOT_DIR}")
     endif(LAPACK_ROOT_DIR)
     find_package(LAPACK REQUIRED)
-    find_library(LAPACKE_LIB NAMES lapacke REQUIRED)
+    find_library(LAPACKE_LIB NAMES openblas REQUIRED)
     set(LAPACK_LIBRARIES ${LAPACKE_LIB} ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES}
                          ${CBLAS_LIBRARIES})
     find_path(
       LAPACKE_INCLUDE_DIRS
       NAMES lapacke.h
-      HINTS ${LAPACK_LIBRARIES})
+      HINTS ${OpenBLAS_INCLUDE_DIRS})
     if(LAPACK_FOUND)
       message(STATUS "Lapacke libraries: ${LAPACK_LIBRARIES}")
       message(STATUS "Lapacke include dirs: ${LAPACKE_INCLUDE_DIRS}")
