--- netcdf_ref/CMakeLists.txt	2022-01-23 21:24:33.029077174 +0100
+++ netcdf_dev/CMakeLists.txt	2022-01-23 21:31:47.176203170 +0100
@@ -750,7 +750,7 @@
     SET(H5_USE_16_API 0)
   ENDIF()
 
-  FIND_PATH(HAVE_HDF5_H hdf5.h)
+  FIND_PATH(HAVE_HDF5_H hdf5.h PATHS ${HDF5_INCLUDE_DIR})
   IF(NOT HAVE_HDF5_H)
     MESSAGE(FATAL_ERROR "Compiling a test with hdf5 failed. Either hdf5.h cannot be found, or the log messages should be checked for another reason.")
   ELSE(NOT HAVE_HDF5_H)
