--- openturns_ref/openturns-1.20.1/CMakeLists.txt	2023-03-31 13:11:11.000000000 +0200
+++ openturns_dev/openturns-1.20.1/CMakeLists.txt	2023-04-05 11:34:07.224075227 +0200
@@ -315,7 +315,11 @@
 endif ()
 
 if (USE_NLOPT)
-  find_package (NLopt CONFIG QUIET)
+  SET(NLOPT_ROOT_DIR $ENV{NLOPT_ROOT_DIR} CACHE PATH "Path to the NLOPT.")
+  IF(NLOPT_ROOT_DIR)
+    LIST(APPEND CMAKE_PREFIX_PATH "${NLOPT_ROOT_DIR}")
+  ENDIF(NLOPT_ROOT_DIR)
+  find_package (NLopt)
   if (NLopt_FOUND)
     message(STATUS "Found NLopt: ${NLopt_DIR}  (found version \"${NLopt_VERSION}\")")
   else ()
