--- openturns/openturns-1.24/lib/test/CMakeLists.txt	2025-04-03 12:53:28.012842479 +0200
+++ openturns.dev/openturns-1.24/lib/test/CMakeLists.txt	2025-04-03 12:57:08.123034928 +0200
@@ -729,7 +729,7 @@
 ot_check_test (LinearLeastSquares_std)
 
 
-add_custom_target (cppcheck COMMAND ${CMAKE_CTEST_COMMAND} --build-config ${CMAKE_BUILD_TYPE} --tests-regex "^cppcheck_"
+add_custom_target (cppcheck COMMAND ${CMAKE_CTEST_COMMAND} --build-config ${CMAKE_BUILD_TYPE} --tests-regex "^cppcheck_" -E "cppcheck_Log_std"
                     DEPENDS ${CHECK_TO_BE_RUN}
                     COMMENT "Run C++ pre-installation tests")
 add_dependencies (check cppcheck)
