diff -Naur openturns-1.20_ref/openturns-1.20/CMakeLists.txt  openturns-1.20_dev/openturns-1.20/CMakeLists.txt
--- openturns-1.20_ref/openturns-1.20/CMakeLists.txt	2023-03-31 13:11:11.000000000 +0200
+++ openturns-1.20_dev/openturns-1.20/CMakeLists.txt	2023-04-04 10:35:09.815050125 +0200
@@ -154,6 +154,10 @@
   endif ()
 endif ()
 list (APPEND OPENTURNS_PRIVATE_LIBRARIES ${LAPACK_LIBRARIES})
+IF (DEFINED CBLAS_LIBRARIES)
+  MESSAGE(STATUS "CBLAS_LIBRARIES is defined... adding to the list of libraries ${CBLAS_LIBRARIES}")
+  LIST (APPEND OPENTURNS_PRIVATE_LIBRARIES ${CBLAS_LIBRARIES})
+ENDIF()
 
 if (USE_SPECTRA)
   find_package (Spectra 1.0 CONFIG)
