--- ParaView-2918b94391_ref/VTK/ThirdParty/vtkm/vtkvtkm/vtk-m/CMake/FindTBB.cmake	2025-04-29 15:07:29.000000000 +0200
+++ ParaView-2918b94391_dev/VTK/ThirdParty/vtkm/vtkvtkm/vtk-m/CMake/FindTBB.cmake	2025-06-24 00:12:08.000000000 +0200
@@ -70,29 +70,29 @@
 
 # Use TBBConfig.cmake if possible.
 
-set(_tbb_find_quiet)
-if (TBB_FIND_QUIETLY)
-  set(_tbb_find_quiet QUIET)
-endif ()
-set(_tbb_find_components)
-set(_tbb_find_optional_components)
-foreach (_tbb_find_component IN LISTS TBB_FIND_COMPONENTS)
-  if (TBB_FIND_REQUIRED_${_tbb_find_component})
-    list(APPEND _tbb_find_components "${_tbb_find_component}")
-  else ()
-    list(APPEND _tbb_find_optional_components "${_tbb_find_component}")
-  endif ()
-endforeach ()
-unset(_tbb_find_component)
-find_package(TBB CONFIG ${_tbb_find_quiet}
-  COMPONENTS ${_tbb_find_components}
-  OPTIONAL_COMPONENTS ${_tbb_find_optional_components})
-unset(_tbb_find_quiet)
-unset(_tbb_find_components)
-unset(_tbb_find_optional_components)
-if (TBB_FOUND)
-  return ()
-endif ()
+# set(_tbb_find_quiet)
+# if (TBB_FIND_QUIETLY)
+#   set(_tbb_find_quiet QUIET)
+# endif ()
+# set(_tbb_find_components)
+# set(_tbb_find_optional_components)
+# foreach (_tbb_find_component IN LISTS TBB_FIND_COMPONENTS)
+#   if (TBB_FIND_REQUIRED_${_tbb_find_component})
+#     list(APPEND _tbb_find_components "${_tbb_find_component}")
+#   else ()
+#     list(APPEND _tbb_find_optional_components "${_tbb_find_component}")
+#   endif ()
+# endforeach ()
+# unset(_tbb_find_component)
+# find_package(TBB CONFIG ${_tbb_find_quiet}
+#   COMPONENTS ${_tbb_find_components}
+#   OPTIONAL_COMPONENTS ${_tbb_find_optional_components})
+# unset(_tbb_find_quiet)
+# unset(_tbb_find_components)
+# unset(_tbb_find_optional_components)
+# if (TBB_FOUND)
+#   return ()
+# endif ()
 
 #====================================================
 # Fix the library path in case it is a linker script
