diff -Naur ParaView-5.6.0-bfaf7b82_SRC/VTK/ThirdParty/vtkm/vtk-m/vtkm/worklet/contourtree_augmented/mesh_dem_meshtypes/CMakeLists.txt ParaView-5.6.0-bfaf7b82_SRC_new/VTK/ThirdParty/vtkm/vtk-m/vtkm/worklet/contourtree_augmented/mesh_dem_meshtypes/CMakeLists.txt
--- ParaView-5.6.0-bfaf7b82_SRC/VTK/ThirdParty/vtkm/vtk-m/vtkm/worklet/contourtree_augmented/mesh_dem_meshtypes/CMakeLists.txt	2019-06-27 14:45:18.725749100 +0300
+++ ParaView-5.6.0-bfaf7b82_SRC_new/VTK/ThirdParty/vtkm/vtk-m/vtkm/worklet/contourtree_augmented/mesh_dem_meshtypes/CMakeLists.txt	2019-06-27 14:18:34.571736200 +0300
@@ -86,7 +86,7 @@
 #1. vtkm_declare_headers will do a flat install into the current respective
 #   build directory
 vtkm_get_kit_name(name dir_prefix)
-vtkm_add_header_build_test(${name} ${dir_prefix} FALSE ${extra_headers})
+#vtkm_add_header_build_test(${name} ${dir_prefix} FALSE ${extra_headers})
 vtkm_install_headers("${dir_prefix}/freudenthal_2D" freudenthal_2D/Types.h)
 vtkm_install_headers("${dir_prefix}/freudenthal_3D" freudenthal_3D/Types.h)
 vtkm_install_headers("${dir_prefix}/marchingcubes_3D" marchingcubes_3D/Types.h)
