--- ParaView-5.11.0-ref/VTK/Rendering/Matplotlib/vtkMatplotlibMathTextUtilities.cxx	2022-11-18 15:26:54.000000000 +0100
+++ ParaView-5.11.0-dev/VTK/Rendering/Matplotlib/vtkMatplotlibMathTextUtilities.cxx	2024-11-27 13:31:31.181887757 +0100
@@ -175,7 +175,7 @@
     return false;
   }
 
-  this->MaskParser = PyObject_CallFunction(mathTextParserClass, const_cast<char*>("s"), "bitmap");
+  this->MaskParser = PyObject_CallFunction(mathTextParserClass, const_cast<char*>("s"), "agg");
   if (this->CheckForError(this->MaskParser))
   {
     Py_CLEAR(this->MaskParser);
@@ -684,7 +684,7 @@
   }
 
   // Get ftimage
-  PyObject* ftImage = PyTuple_GetItem(resTupleParse, 0);
+  PyObject* ftImage = PyTuple_GetItem(resTupleParse, 5);
   if (this->CheckForError(ftImage))
   {
     return false;
@@ -790,6 +790,7 @@
 //------------------------------------------------------------------------------
 bool vtkMatplotlibMathTextUtilities::SetMathTextFont(vtkTextProperty* tprop)
 {
+  vtkPythonScopeGilEnsurer gilEnsurer;
   vtkSmartPyObject mplBase(PyImport_ImportModule("matplotlib"));
   if (this->CheckForError(mplBase))
   {
