--- ParaView_ref/ThirdParty/catalyst/vtkcatalyst/catalyst/CMakeLists.txt	2021-02-12 16:35:40.957930618 +0100
+++ ParaView/ThirdParty/catalyst/vtkcatalyst/catalyst/CMakeLists.txt	2021-02-12 16:35:50.500219714 +0100
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.13...3.16 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.12...3.16 FATAL_ERROR)
 project(CATALYST VERSION 2.0 LANGUAGES C CXX)
 
 #------------------------------------------------------------------------------
