--- paravis_ref/src/Plugins/MEDWriter/plugin/Test/TestMEDWriter0.py	2025-07-07 11:46:57.319355225 +0200
+++ paravis_dev/src/Plugins/MEDWriter/plugin/Test/TestMEDWriter0.py	2025-07-07 10:59:32.681666936 +0200
@@ -22,12 +22,13 @@
 from paraview.simple import *
 import MEDLoader as ml
 import os
+import platform
 from math import pi,sqrt
 import tempfile
 
 #### disable automatic camera reset on 'Show'
 paraview.simple._DisableFirstRenderCameraReset()
-
+currentDir = os.getcwd()
 with tempfile.TemporaryDirectory(prefix="MEDWriter_") as tmpdir:
     pat=os.path.join(tmpdir, 'testMEDWriter_%i.med')
     fname0=pat%0
@@ -249,4 +250,7 @@
     assert(f11_array_double.isEqualWithoutConsideringStr(f10.getArray(),1e-12))
     f11_n_array_double = f11_n.getArray().convertToDblArr() # f11_n is MEDCouplingFieldFloat
     assert(f11_n_array_double.isEqualWithoutConsideringStr(f10_n.getArray(),1e-12))
-
+    # On Windows, if one deletes the temporary directory while busy
+    # with it, it raises an error. So we change back to the original
+    if platform.system() == "Windows" :
+        os.chdir(currentDir)
