--- paraview-5.9.0_ref/VTK/Utilities/Python/vtkPython.h	2022-04-19 10:11:34.466378625 +0200
+++ paraview-5.9.0_new/VTK/Utilities/Python/vtkPython.h	2022-04-19 10:13:54.217617713 +0200
@@ -76,8 +76,12 @@
 #define HAVE_ROUND 1
 #endif
 
-#include <Python.h>
-
+//#include <Python.h>
+#ifdef VTK_PYTHON_UNDEF_DEBUG
+ #define _DEBUG 1
+ #include <Python.h>
+ #undef _DEBUG
+#endif
 #if defined(_MSC_VER) && _MSC_VER >= 1800
 #undef HAVE_ROUND
 #endif
