--- ParaView-6.1.0-c61dcc8ee0_ref/VTK/Filters/ONNX/vtkONNXInference.cxx	2025-10-23 22:57:05.000000000 +0200
+++ ParaView-6.1.0-c61dcc8ee0_dev/VTK/Filters/ONNX/vtkONNXInference.cxx	2025-11-07 13:25:08.732169800 +0100
@@ -13,6 +13,11 @@
 #include "vtkStreamingDemandDrivenPipeline.h"
 
 #include <onnxruntime_cxx_api.h>
+#ifdef _WIN32
+#include <Windows.h>
+#include <string>
+#endif
+
 
 VTK_ABI_NAMESPACE_BEGIN
 
@@ -47,6 +52,30 @@
   os << indent << "InputSize: " << this->InputSize << endl;
 }
 
+
+#if defined(WIN32)
+  // these two functions are extracted from Basics_Utils.cxx SALOME
+  // they are similar to those in Basics_Utils.cxx but avoid dependency to SALOME
+  // Convert an UTF8 string to a wide Unicode String
+  wchar_t* f_utf8_decode(const char* decoded)
+  {
+	  if (decoded == NULL) return NULL;
+    auto size_needed = MultiByteToWideChar(CP_UTF8, 0, decoded, (int)strlen(decoded), NULL, 0);
+	  wchar_t* wstrTo = new wchar_t[ size_needed + 1 ];
+	  MultiByteToWideChar(CP_UTF8, 0, decoded, (int)strlen(decoded), wstrTo, size_needed);
+	  wstrTo[size_needed] = '\0';
+	  return wstrTo;
+  }
+
+  // Convert an UTF8 std::string to a wide Unicode std::wstring
+  std::wstring f_utf8_decode_s(const std::string& decoded) {
+	  wchar_t* aDecoded = f_utf8_decode(decoded.c_str());
+	  std::wstring res = std::wstring(aDecoded);
+	  delete [] aDecoded;
+	  return res;
+  }
+#endif // WIN32
+
 //------------------------------------------------------------------------------
 void vtkONNXInference::InitializeSession()
 {
@@ -68,8 +97,14 @@
 
   try
   {
+#if defined(WIN32)
+    std::wstring modelPath_ws = f_utf8_decode_s(this->ModelFile);
+    this->Internals->Session = std::make_unique<Ort::Session>(
+      this->Internals->OrtEnv, modelPath_ws.c_str(), sessionOptions);
+#else
     this->Internals->Session = std::make_unique<Ort::Session>(
       this->Internals->OrtEnv, this->ModelFile.c_str(), sessionOptions);
+#endif // WIN32
   }
   catch (const Ort::Exception& e)
   {
