diff --git a/src/SHAPERGUI/resources/LightApp.xml.in b/src/SHAPERGUI/resources/LightApp.xml.in
index 081f2b03e..c86d74b96 100644
--- a/src/SHAPERGUI/resources/LightApp.xml.in
+++ b/src/SHAPERGUI/resources/LightApp.xml.in
@@ -22,6 +22,9 @@
     <parameter name="PartSet" value="%SHAPER_ROOT_DIR%/share/salome/resources/shaper"/>
     <parameter name="XGUI" value="%SHAPER_ROOT_DIR%/share/salome/resources/shaper"/>
   </section>
+  <section name="Sketch">
+    <parameter name="operation_cursor" value="1"/>
+  </section>
   <section name="Viewer" >
     <!-- Viewer preferences -->
     <parameter name="face-selection" value="true" />
