--- persalys_v18.0-ref/CMakeLists.txt	2025-04-02 16:11:14.000000000 +0200
+++ persalys_v18.0-dev/CMakeLists.txt	2025-04-07 09:58:30.000000000 +0200
@@ -64,7 +64,7 @@
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj")
 endif ()
 
-find_package (Boost CONFIG COMPONENTS locale program_options)
+find_package (Boost COMPONENTS locale filesystem  program_options REQUIRED)
 if (Boost_FOUND)
   message(STATUS "Found Boost: ${Boost_DIR} (found suitable version \"${Boost_VERSION}\")")
 else ()
@@ -108,6 +108,7 @@
 endif ()
 
 if (USE_SALOME)
+  find_package (SalomeBootstrap CONFIG)
   find_package (SalomeKERNEL CONFIG)
   find_package (Qt5WebEngine CONFIG)
   find_package (Qt5WebEngineWidgets CONFIG)
@@ -115,6 +116,12 @@
   find_package (SalomeYACS CONFIG)
   find_package (Py2cpp CONFIG)
   find_package (ydefx CONFIG)
+
+  if (SalomeBootstrap_FOUND)
+    list (APPEND PERSALYS_DEFINITIONS ${SALOMEBOOTSTRAP_DEFINITIONS})
+    include_directories(${SALOMEBOOTSTRAP_INCLUDE_DIRS})
+  endif()
+
   if (SalomeYACS_FOUND AND ydefx_FOUND AND SalomeGUI_FOUND)
     list (APPEND PERSALYS_DEFINITIONS ${YACS_DEFINITIONS})
     list (APPEND PERSALYS_DEFINITIONS -DPERSALYS_HAVE_YACS)
