--- persalys_v19.0_ref/CMakeLists.txt	2025-04-02 16:11:14.000000000 +0200
+++ persalys_v19.0_dev/CMakeLists.txt	2025-04-04 11:01:57.219684262 +0200
@@ -108,6 +108,7 @@
 endif ()
 
 if (USE_SALOME)
+  find_package (SalomeBootstrap CONFIG)
   find_package (SalomeKERNEL CONFIG)
   find_package (Qt5WebEngine CONFIG)
   find_package (Qt5WebEngineWidgets CONFIG)
@@ -115,6 +116,12 @@
   find_package (SalomeYACS CONFIG)
   find_package (Py2cpp CONFIG)
   find_package (ydefx CONFIG)
+
+  if (SalomeBootstrap_FOUND)
+    list (APPEND PERSALYS_DEFINITIONS ${SALOMEBOOTSTRAP_DEFINITIONS})
+    include_directories(${SALOMEBOOTSTRAP_INCLUDE_DIRS})
+  endif()
+
   if (SalomeYACS_FOUND AND ydefx_FOUND AND SalomeGUI_FOUND)
     list (APPEND PERSALYS_DEFINITIONS ${YACS_DEFINITIONS})
     list (APPEND PERSALYS_DEFINITIONS -DPERSALYS_HAVE_YACS)
