--- persalys_v18.0-ref/lib/src/base/CMakeLists.txt	2025-04-02 16:11:14.000000000 +0200
+++ persalys_v18.0-dev/lib/src/base/CMakeLists.txt	2025-04-07 09:15:32.000000000 +0200
@@ -234,10 +234,9 @@
 if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1)
   target_link_libraries (persalysbase PRIVATE stdc++fs)
 endif ()
-
-target_link_libraries (persalysbase PUBLIC ${PERSALYS_PUBLIC_LIBRARIES})
-target_link_libraries (persalysbase PRIVATE ${PERSALYS_PRIVATE_LIBRARIES})
-target_link_libraries (persalysbase PRIVATE Boost::locale)
+target_compile_options(persalysbase PUBLIC  -DBOOST_ALL_DYN_LINK)
+target_link_libraries (persalysbase PUBLIC "${PERSALYS_PUBLIC_LIBRARIES};${Boost_LIBRARIES};${Boost_FILESYSTEM_LIBRARY_RELEASE}")
+target_link_libraries (persalysbase PRIVATE "${PERSALYS_PRIVATE_LIBRARIES};${Boost_LIBRARIES};${Boost_FILESYSTEM_LIBRARY_RELEASE}")
 target_link_libraries (persalysbase PRIVATE Python::Python)
 
 install (TARGETS persalysbase
