--- planegcs-0.20_ref/planegcs/GCS.cpp	2022-09-30 13:30:52.206867543 +0200
+++ planegcs-0.20_dev/planegcs/GCS.cpp	2022-09-30 14:03:08.348487061 +0200
@@ -95,14 +95,14 @@
 #define CASE_NOT_IMP(X) case X: { subsystemfile << "//" #X "not yet implemented" << std::endl; break; }
 #endif
 
-#include <FCConfig.h>
+//#include <FCConfig.h>
 #ifdef _GCS_USE_STL_OUTPUT
 #include <cstdio>
 #else
 #include <Base/Console.h>
 #endif
 
-#include <boost_graph_adjacency_list.hpp>
+#include <boost/graph/adjacency_list.hpp>
 #include <boost/graph/connected_components.hpp>
 
 namespace GCS
