diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ec9d1e8..1ca0b92 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -46,12 +46,6 @@ SET(_py2cpp_headers
   py2cpp.hxx
 )
 
-if(WIN32)
-  add_library(py2cpp_lib STATIC  ${_py2cpp_sources})
-  TARGET_LINK_LIBRARIES(py2cpp_lib ${_link_LIBRARIES})
-  set_target_properties(py2cpp_lib PROPERTIES OUTPUT_NAME py2cpp)
-endif(WIN32)
-
 ADD_LIBRARY(py2cpp  ${_py2cpp_sources})
 TARGET_LINK_LIBRARIES(py2cpp ${_link_LIBRARIES})
 TARGET_INCLUDE_DIRECTORIES(py2cpp PUBLIC
@@ -59,8 +53,9 @@ TARGET_INCLUDE_DIRECTORIES(py2cpp PUBLIC
                           $<INSTALL_INTERFACE:include>)
 
 IF(WIN32)
-  INSTALL(TARGETS py2cpp EXPORT Py2cppCfg LIBRARY DESTINATION lib ${CMAKE_INSTALL_LIBDIR}  RUNTIME DESTINATION bin ${CMAKE_INSTALL_BINDIR})
-  INSTALL(TARGETS py2cpp_lib DESTINATION lib)
+  INSTALL(TARGETS py2cpp EXPORT Py2cppCfg
+	ARCHIVE DESTINATION lib
+	RUNTIME DESTINATION bin)
 ELSE(WIN32)
   INSTALL(TARGETS py2cpp EXPORT Py2cppCfg LIBRARY DESTINATION lib)
 ENDIF(WIN32)
