--- PyFMI-2.6_ref/src/pyfmi/fmi_coupled.pyx	2024-01-15 10:37:54.999574589 +0100
+++ PyFMI-2.6_dev/src/pyfmi/fmi_coupled.pyx	2024-01-15 10:37:43.735520048 +0100
@@ -1004,7 +1004,7 @@
         Calls the low-level FMI function: fmi2SetReal
         """
         input_valueref = np.array(valueref, ndmin=1).ravel()
-        set_value      = np.array(values, dtype=np.float, ndmin=1).ravel()
+        set_value      = np.array(values, dtype=np.float32, ndmin=1).ravel()
 
         if input_valueref.size != set_value.size:
             raise fmi.FMUException('The length of valueref and values are inconsistent.')
