--- Python-3.9.14-ref/Lib/ctypes/__init__.py	2022-09-06 19:26:16.000000000 +0200
+++ Python-3.9.14-new/Lib/ctypes/__init__.py	2024-04-03 17:34:16.000000000 +0200
@@ -364,7 +364,14 @@
                 if '/' in name or '\\' in name:
                     self._name = nt._getfullpathname(self._name)
                     mode |= nt._LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR
-
+                # PATH is how DLL lookup has always worked in the past
+                # in Python on Windows. IMHO both the above mode flags
+                # are not wanted and cause many serious regressions within
+                # the conda ecosystem on Windows. We should however
+                # propagate any PATH changes that have happened to Python
+                # library and that is not yet implemented.
+                LOAD_WITH_ALTERED_SEARCH_PATH = 0x00000008
+                mode = LOAD_WITH_ALTERED_SEARCH_PATH
         class _FuncPtr(_CFuncPtr):
             _flags_ = flags
             _restype_ = self._func_restype_
