diff -Naur qt.ref/qtbase/src/corelib/global/qendian.h qt.dev/qtbase/src/corelib/global/qendian.h
--- qt.ref/qtbase/src/corelib/global/qendian.h	2020-09-21 15:16:21.000000000 +0200
+++ qt.dev/qtbase/src/corelib/global/qendian.h	2023-10-11 10:52:55.992773784 +0200
@@ -47,6 +47,7 @@
 // include stdlib.h and hope that it defines __GLIBC__ for glibc-based systems
 #include <stdlib.h>
 #include <string.h>
+#include <limits>
 
 #ifdef min // MSVC
 #undef min
diff -Naur qt.ref/qtbase/src/corelib/tools/qbytearraymatcher.h qt.dev/qtbase/src/corelib/tools/qbytearraymatcher.h
--- qt.ref/qtbase/src/corelib/tools/qbytearraymatcher.h	2020-09-21 15:16:21.000000000 +0200
+++ qt.dev/qtbase/src/corelib/tools/qbytearraymatcher.h	2023-10-11 10:57:35.066370201 +0200
@@ -41,6 +41,7 @@
 #define QBYTEARRAYMATCHER_H
 
 #include <QtCore/qbytearray.h>
+#include <limits>
 
 QT_BEGIN_NAMESPACE
 
diff -Naur qt.ref/qtdeclarative/src/qmldebug/qqmlprofilerevent_p.h qt.dev/qtdeclarative/src/qmldebug/qqmlprofilerevent_p.h
--- qt.ref/qtdeclarative/src/qmldebug/qqmlprofilerevent_p.h	2020-10-15 08:53:19.000000000 +0200
+++ qt.dev/qtdeclarative/src/qmldebug/qqmlprofilerevent_p.h	2023-10-11 11:36:43.775903667 +0200
@@ -50,6 +50,8 @@
 #include <initializer_list>
 #include <type_traits>
 
+#include <limits>
+
 //
 //  W A R N I N G
 //  -------------
