diff -Naur rapidjson/include/rapidjson/document.h rapidjson.dev/include/rapidjson/document.h
--- rapidjson/include/rapidjson/document.h	2016-08-25 09:59:05.000000000 +0200
+++ rapidjson.dev/include/rapidjson/document.h	2025-03-05 10:29:40.526677466 +0100
@@ -316,8 +316,6 @@
 
     GenericStringRef(const GenericStringRef& rhs) : s(rhs.s), length(rhs.length) {}
 
-    GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
-
     //! implicit conversion to plain CharType pointer
     operator const Ch *() const { return s; }
 
@@ -328,6 +326,8 @@
     //! Disallow construction from non-const array
     template<SizeType N>
     GenericStringRef(CharType (&str)[N]) /* = delete */;
+     //! Copy assignment operator not permitted - immutable type
+    GenericStringRef& operator=(const GenericStringRef& rhs) /* = delete */;
 };
 
 //! Mark a character pointer as constant string
