diff --git a/cmake/FindXDR.cmake b/cmake/FindXDR.cmake
index e8e8625..52f2de9 100644
--- a/cmake/FindXDR.cmake
+++ b/cmake/FindXDR.cmake
@@ -19,7 +19,7 @@
 
 MESSAGE(STATUS "Check for XDR ...")
 
-FIND_PATH(XDR_INCLUDE_DIRS rpc/xdr.h)
+FIND_PATH(XDR_INCLUDE_DIRS rpc/xdr.h PATH_SUFFIXES tirpc)
 IF(XDR_INCLUDE_DIRS)
    SET(XDR_DEFINITIONS "-DHAS_XDR")
 ENDIF()
@@ -34,6 +34,14 @@ IF(WIN32)
       SET(XDR_INCLUDE_DIRS "${XDR_INCLUDE_DIRS2}")  # Make the detection fail
     ENDIF()
   ENDIF()
+ELSE(WIN32)
+  FIND_LIBRARY(XDR_LIBRARIES NAMES tirpc xdr)
+  IF(NOT XDR_LIBRARIES)
+    MESSAGE(STATUS  "Could not find XDR libraries ...")
+    SET(XDR_LIBRARIES) # reset variable tough
+  ELSE()
+    MESSAGE(STATUS "Found XDR libraries ${XDR_LIBRARIES} ...")
+  ENDIF()
 ENDIF(WIN32)
 
 INCLUDE(FindPackageHandleStandardArgs)
