diff --git a/cmake/SalomeMacros.cmake b/cmake/SalomeMacros.cmake
index bc7baaa..6fa6fe0 100755
--- a/cmake/SalomeMacros.cmake
+++ b/cmake/SalomeMacros.cmake
@@ -120,10 +120,11 @@ MACRO(SALOME_INSTALL_SCRIPTS file_list path)
     SET(_source_prefix "")
     GET_FILENAME_COMPONENT(file_name ${file} NAME)
     IF(NOT IS_ABSOLUTE ${file})
+      SET(_source_prefix "${CMAKE_CURRENT_SOURCE_DIR}/")
       IF(SALOME_INSTALL_SCRIPTS_WORKING_DIRECTORY)
         SET(PREFIX "${SALOME_INSTALL_SCRIPTS_WORKING_DIRECTORY}/")
+        SET(_source_prefix "${SALOME_INSTALL_SCRIPTS_WORKING_DIRECTORY}/")
       ENDIF(SALOME_INSTALL_SCRIPTS_WORKING_DIRECTORY)
-      SET(_source_prefix "${CMAKE_CURRENT_SOURCE_DIR}/")
     ENDIF(NOT IS_ABSOLUTE ${file})
     INSTALL(FILES ${PREFIX}${file} DESTINATION ${path} PERMISSIONS ${PERMS})
     GET_FILENAME_COMPONENT(ext ${file} EXT)
