diff --git a/src/LightApp/resources/LightApp_msg_en.ts b/src/LightApp/resources/LightApp_msg_en.ts
index bb72f990a..ab8bc2223 100644
--- a/src/LightApp/resources/LightApp_msg_en.ts
+++ b/src/LightApp/resources/LightApp_msg_en.ts
@@ -9,14 +9,11 @@
     </message>
     <message>
         <source>ABOUT_COPYRIGHT</source>
-        <translation>Copyright (C) 2007-2023 CEA, EDF, OPEN CASCADE
-
-Copyright (C) 2003-2007 OPEN CASCADE, EADS/CCR, LIP6,
-CEA/DEN, CEDRAT, EDF R&amp;D, LEG, PRINCIPIA R&amp;D, BUREAU VERITAS</translation>
+        <translation>Copyright (C) 2007-2024 CEA</translation>
     </message>
     <message>
         <source>ABOUT_CAPTION</source>
-        <translation>About %1</translation>
+        <translation>About ALAMOS</translation>
     </message>
     <message>
         <source>ABOUT_BASE</source>
@@ -38,10 +35,6 @@ CEA/DEN, CEDRAT, EDF R&amp;D, LEG, PRINCIPIA R&amp;D, BUREAU VERITAS</translatio
         <source>ABOUT_CLOSE</source>
         <translation>&amp;Close</translation>
     </message>
-    <message>
-        <source>APP_NAME</source>
-        <translation>SALOME</translation>
-    </message>
     <message>
         <source>ABOUT_LICENSE</source>
         <translation>GNU LGPL</translation>
diff --git a/src/LightApp/resources/LightApp_msg_fr.ts b/src/LightApp/resources/LightApp_msg_fr.ts
index 00b1bf008..72f846834 100644
--- a/src/LightApp/resources/LightApp_msg_fr.ts
+++ b/src/LightApp/resources/LightApp_msg_fr.ts
@@ -9,10 +9,7 @@
     </message>
     <message>
         <source>ABOUT_COPYRIGHT</source>
-        <translation>Copyright (C) 2007-2023 CEA, EDF, OPEN CASCADE
-
-Copyright (C) 2003-2007 OPEN CASCADE, EADS/CCR, LIP6,
-CEA/DEN, CEDRAT, EDF R&amp;D, LEG, PRINCIPIA R&amp;D, BUREAU VERITAS</translation>
+        <translation>Copyright (C) 2007-2024 CEA</translation>
     </message>
     <message>
         <source>ABOUT_CAPTION</source>
@@ -38,10 +35,6 @@ CEA/DEN, CEDRAT, EDF R&amp;D, LEG, PRINCIPIA R&amp;D, BUREAU VERITAS</translatio
         <source>ABOUT_CLOSE</source>
         <translation>&amp;Fermer</translation>
     </message>
-    <message>
-        <source>APP_NAME</source>
-        <translation>SALOME</translation>
-    </message>
     <message>
         <source>ABOUT_LICENSE</source>
         <translation>GNU LGPL</translation>
diff --git a/src/LightApp/resources/LightApp_msg_ja.ts b/src/LightApp/resources/LightApp_msg_ja.ts
index 99841e9d7..77421ee15 100644
--- a/src/LightApp/resources/LightApp_msg_ja.ts
+++ b/src/LightApp/resources/LightApp_msg_ja.ts
@@ -9,10 +9,7 @@
     </message>
     <message>
       <source>ABOUT_COPYRIGHT</source>
-      <translation>Copyright (C) 2007-2023 CEA, EDF, OPEN CASCADE
-
-Copyright (C) 2003-2007 OPEN CASCADE, EADS/CCR, LIP6,
-CEA/DEN, CEDRAT, EDF R&amp;D, LEG, PRINCIPIA R&amp;D, BUREAU VERITAS</translation>
+      <translation>Copyright (C) 2007-2024 CEA</translation>
     </message>
     <message>
       <source>ABOUT_CAPTION</source>
@@ -38,10 +35,6 @@ CEA/DEN, CEDRAT, EDF R&amp;D, LEG, PRINCIPIA R&amp;D, BUREAU VERITAS</translatio
       <source>ABOUT_CLOSE</source>
       <translation>閉じる(&amp;C)</translation>
     </message>
-    <message>
-      <source>APP_NAME</source>
-      <translation>SALOME</translation>
-    </message>
     <message>
       <source>ABOUT_LICENSE</source>
       <translation>GNU LGPL</translation>
