diff -Naur numpy-1.22.2_ref/setup.py numpy-1.22.2_dev/setup.py
--- numpy-1.22.2_ref/setup.py	2022-09-06 12:16:24.000000000 +0200
+++ numpy-1.22.2_dev/setup.py	2022-09-06 14:20:07.476055406 +0200
@@ -46,7 +46,7 @@
 # Needed for backwards code compatibility below and in some CI scripts.
 # The version components are changed from ints to strings, but only VERSION
 # seems to matter outside of this module and it was already a str.
-FULLVERSION = versioneer.get_version()
+FULLVERSION = '1.22.2'
 
 # Capture the version string:
 # 1.22.0.dev0+ ... -> ISRELEASED == False, VERSION == 1.22.0
